File: src/main/java/com/github/yukulab/blockhideandseekmod/command/Rules.java
package com.github.yukulab.blockhideandseekmod.command;

import com.github.yukulab.blockhideandseekmod.screen.RuleBookScreen;
import com.mojang.brigadier.Command;
import net.fabricmc.fabric.api.command.v1.CommandRegistrationCallback;

import static net.minecraft.server.command.CommandManager.literal;

public class Rules {
    public static void registerCommands() {
        CommandRegistrationCallback.EVENT.register(
                (dispatcher, dedicated) ->
                        dispatcher.register(literal("bhas")
                                .then(literal("rules")
                                        .executes(context -> {
                                            var source = context.getSource();
                                            RuleBookScreen.open(source.getPlayer());
                                            return Command.SINGLE_SUCCESS;
                                        })
                                )
                        )
        );
    }
}

File: src/main/java/com/github/yukulab/blockhideandseekmod/command/Rules.kt
package com.github.yukulab.blockhideandseekmod.command

import com.github.yukulab.blockhideandseekmod.item.BhasItems
import dev.uten2c.strobo.command.CommandBuilder
import net.minecraft.item.ItemStack

object Rules : BHASCommand {
    override val builder: CommandBuilder.() -> Unit = {
        literal("rules") {
            executes {
                player.inventory.insertStack(ItemStack(BhasItems.RuleBook))
            }

            players { getPlayer ->
                requires {
                    it.hasPermissionLevel(it.server.opPermissionLevel)
                }
                executes {
                    getPlayer().forEach {
                        it.inventory.insertStack(ItemStack(BhasItems.RuleBook))
                    }
                }
            }
        }
    }
}